### DISCUSSION TITLE
How does the API to start a dynamic workflow work?

### DISCUSSION THREAD

**Discussion by pallavisantosh19** [Netflix/conductor#3230] [Q&A]
Per [this document ](https://conductor.netflix.com/gettingstarted/startworkflow.html), a workflow can be started either by name (if it is predefined) or by providing a definition at runtime (dynamic workflow).

However, I've found that if you provide a new workflow definition in this start workflow API with the name of a pre-registered workflow, it simply overrides the existing pre-defined definition with the new definition provided in the request body for that execution (rather than erroring out because you are attempting to define and start a new workflow with a name that is already taken).

I haven't found any documentation that states that this is the expected behavior for the API - that you can either start an existing workflow, start a new workflow or **override fields in an existing workflow for the execution and then start it**.

Is this an accurate understanding of how this API is intended to work?

**Comment by apanicker-nflx** [comment]
Yes, the workflow definition that is provided as part of the stat workflow request takes precedence over the registered workflow definition. This is intentional because the user is explicitly providing the workflow definition that they would like to run instead of using a pre-registered one.
